### PR TITLE
amount input switchable to usd

### PR DESCRIPTION
**Pull request by futurepaul**
Screenshot 2023-05-31 at 4 38 45 PM

### CODE REVIEW

File: src/components/AmountEditable.tsx [hunk 1]
```diff
@@ -204,48 +261,66 @@ export const AmountEditable: ParentComponent<{
...
+                onInput={handleSatsInput}
+              />
+              <label classList={{ "bg-pink-500": mode() === "fiat" }}>Fiat</label>
+              <pre>{displayFiat()} USD</pre>
+              <input
+                ref={(el) => (fiatInputRef = el)}
+                disabled={mode() === "sats"}
+                type="text"
+                class="bg-black text-white"
+                // class="opacity-0 absolute -z-10"
```
Comment (benthecarman) [inline src/components/AmountEditable.tsx:284]: nit: can remove this i think

**Review by benthecarman — CHANGES_REQUESTED** [review]
The fiat to sats conversion doesn't seem to work anymore

image

**Comment by benthecarman** [comment]
Also not sure I love the way to switch between fiat and sats, normally you would just click the fiat number. I only really knew the swap button was the button for this because I was reviewing the PR

File: src/components/AmountEditable.tsx (hunk 1 above)
Comment (futurepaul) [reply src/components/AmountEditable.tsx]: thx

**Comment by futurepaul** [comment]
okay updated with tap the amount to switch... also the price api seems to have been weird recently, can you try again?

btw you have to start on the home screen and then go to the receive screen to get price data... not exactly sure why. I checked and the same thing is true for the existing amount input screen

**Comment by TonyGiorgio** [comment]
When I type in a few numbers, tap the switch and try typing more numbers, it does not work.

**Comment by futurepaul** [comment]
> When I type in a few numbers, tap the switch and try typing more numbers, it does not work.

okay this was a very helpful comment I realized I can just always let the person keep typing in fiat view and just shift the decimal... before I was deleting anything the user typed past two decimals which is what you were seeing when you entered some sats then switched to fiat

**Comment by TonyGiorgio** [comment]
Looks and works great!

**Review by benthecarman — APPROVED** [review]

**Pull request merged**